app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -74,48 +74,29 @@ private void storeStuct(IASTDeclaration[] declarations, String name) throws Exce
             }
         }
         structRepo.put(name, struct);
-    }
 
-    public void mapPayload(IASTNode[] comments) {
+    }
 
+    private void associateIDStruct(IASTNode[] comments) {
+        Pattern pattern = Pattern.compile("@canPayloadStruct\\s+(\\S+)\\s*=\\s*(\\S+)");
         for (IASTNode comment : comments) {
-            char[] line = simplifyString(comment.getRawSignature());
-            if (line[0] == '|') {
-                String idName = findString(line, 1);
-                int secondIndex = idName.length() + 2;
-                String structName = findString(line, secondIndex);
-                if (!idStruct.containsKey(structName)) {
-                    PayloadMap rep = new PayloadMap(structName);
-                    idStruct.put(structName, rep);
+            Matcher matcher = pattern.matcher(comment.getRawSignature());
+            if (matcher.find()) {
+                if (!idStruct.containsKey(matcher.group(2))) {
+                    ArrayList<String> list = new ArrayList<>();
+                    idStruct.put(matcher.group(2), list);
```

Comment:
```suggestion
                    idStruct.put(matcher.group(2), new ArrayList<>());
```

I don't think having the intermediary `list` is too useful?